js/api.js

```diff
@@ -0,0 +1,5 @@
+async function fetchApi(fetchURL) {
```

Comment:
nice idea to use you own fetch fucntion, but you should implement deafult error handling inside the function

Reply:
good improvements.